### PR TITLE
bleOnDemand implementation

### PR DESCRIPTION
**Pull request by bibarrav**
Hi!,
Here an OnDemand implementation!, to send commands with connections on demand and a new command status, to get light/position/battery on demand too.

Please, take a look!

All the best,
BIV

### CODE REVIEW

**Comment by bibarrav** [comment]
Hi!,
I added the docs for ondemand mode :)

All the best,
BIV

File: src/AM43Client.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ class AM43Client : public BLEClientCallbacks {
     boolean m_DoConnect;
     // We're logged in (correct pin)
     boolean m_LoggedIn;
+    // OnDemand command flag
+    bollean m_CommandQueued;
```
Comment (T-REX-XP) [inline src/AM43Client.h:57]: Please fix the misprint:  boolean instead of bollean

**Review by T-REX-XP — COMMENTED** [review]
I have left a review comment. Please fix it

File: src/AM43Client.h (hunk 1 above)
Comment (bibarrav) [reply src/AM43Client.h]: Ok, done!

**Comment by bibarrav** [comment]
@T-REX-XP @buxtronix Its good to go?

**Comment by buxtronix** [comment]
Yep, just got around to testing it.

**Pull request merged**